### PR TITLE
merged with mesh version  0.6.2 from nordic

### PR DESCRIPTION
**Pull request by AlexDM0**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -125,7 +125,7 @@ ENDIF()
 LIST(APPEND FOLDER_SOURCE "${SOURCE_DIR}/events/cs_Dispatcher.cpp")
 
 # should be only when creating iBeacon
-LIST(APPEND FOLDER_SOURCE "${SOURCE_DIR}/cs_iBeacon.cpp")
```
Comment (eggerdo) [inline CMakeLists.txt]: why did you remove cs_iBeacon?

File: include/third/protocol/mesh_srv.h [hunk 2]
```diff
@@ -39,6 +39,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "trickle.h"
 #include "ble.h"
 #include <stdint.h>
+#include <stdbool.h>
```
Comment (eggerdo) [inline include/third/protocol/mesh_srv.h]: not needed

File: include/third/protocol/timeslot_handler.h [hunk 3]
```diff
@@ -39,6 +39,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "radio_control.h"
 #include "timer_control.h"
 #include <stdint.h>
+#include <stdbool.h>
```
Comment (eggerdo) [inline include/third/protocol/timeslot_handler.h]: not needed

File: src/third/protocol/timeslot_handler.c [hunk 4]
```diff
@@ -119,6 +119,7 @@ static uint32_t g_timeslot_end_timer;
 static uint32_t g_next_timeslot_length;    
 static uint32_t g_start_time_ref = 0;     
 static uint32_t g_is_in_timeslot = false; 
+static uint32_t g_framework_initialized = false;
```
Comment (eggerdo) [inline src/third/protocol/timeslot_handler.c]: make it a boolean

File: src/third/protocol/transport_control.c [hunk 5]
```diff
@@ -288,8 +288,8 @@ void transport_control_timeslot_begin(uint32_t global_timer_value)
 
 void transport_control_step(void)
 {
-    uint32_t next_time;
-    uint32_t time_now = timer_get_timestamp() + global_time;
+    uint64_t next_time;
+    uint64_t time_now = timer_get_timestamp() + global_time;
     trickle_time_update(time_now);
     uint32_t error_code = mesh_srv_get_next_processing_time(&next_time);
```
Comment (eggerdo) [inline src/third/protocol/transport_control.c]: mesh_srv_get_next_processing_time still takes a uint32_t as parameter. mistake?

File: src/third/protocol/trickle.c [hunk 6]
```diff
@@ -52,10 +52,10 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 *****************************************************************************/
 
 
-static uint32_t g_trickle_time; /* global trickle time that all time variables are relative to */
+static uint64_t g_trickle_time; /* global trickle time that all time variables are relative to */
 
 static uint8_t rng_vals[64];
-static uint8_t rng_index = 0;
+static uint8_t rng_index;
```
Comment (eggerdo) [inline src/third/protocol/trickle.c]: why did you remove variable initialization? It's safer to init

File: CMakeLists.txt (hunk 1 above)
Comment (AlexDM0) [inline CMakeLists.txt]: because there was no option for it in the config and I didnt need it. It shouldnt have been added to the push tho, my bad.

File: include/third/protocol/mesh_srv.h (hunk 2 above)
Comment (AlexDM0) [inline include/third/protocol/mesh_srv.h]: ok, just merged in everything from the nordic branch

File: src/third/protocol/timeslot_handler.c (hunk 4 above)
Comment (AlexDM0) [inline src/third/protocol/timeslot_handler.c]: good point.

File: src/third/protocol/transport_control.c (hunk 5 above)
Comment (AlexDM0) [inline src/third/protocol/transport_control.c]: yes has to be 64.. I'll fix it

File: src/third/protocol/trickle.c (hunk 6 above)
Comment (AlexDM0) [inline src/third/protocol/trickle.c]: merged it from nordic like this.

File: src/third/protocol/transport_control.c (hunk 5 above)
Comment (eggerdo) [inline src/third/protocol/transport_control.c]: you need to update the function header in mesh_srv.h too ;)

**Comment by AlexDM0** [comment]
the 64 bit workaround seems to be too much guessing. For now back to vanilla version from Nordic.
